packages/data/src/components/with-dispatch/test/index.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import TestRenderer from 'react-test-renderer';
+
+/**
+ * WordPress dependencies
+ */
+import { createElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import withDispatch from '../';
+import { createRegistry } from '../../../registry';
+import RegistryProvider from '../../registry-provider';
+
+describe( 'withDispatch', () => {
+	let registry;
+	beforeEach( () => {
+		registry = createRegistry();
+	} );
+
+	it( 'passes the relevant data to the component', () => {
+		const store = registry.registerStore( 'counter', {
+			reducer: ( state = 0, action ) => {
+				if ( action.type === 'increment' ) {
+					return state + action.count;
+				}
+				return state;
+			},
+			actions: {
+				increment: ( count = 1 ) => ( { type: 'increment', count } ),
+			},
+		} );
+
+		const Component = withDispatch( ( _dispatch, ownProps ) => {
+			const { count } = ownProps;
+
+			return {
+				increment: () => _dispatch( 'counter' ).increment( count ),
+			};
+		} )( ( props ) => <button onClick={ props.increment } /> );
+
+		const testRenderer = TestRenderer.create(
```

Comment:
My own notes: to replace `shallow` from `enzyme` we would need only:
```js
jest.shallow = ( element ) => {
    const TestRenderer  = require( 'react-test-renderer' );
    const testRenderer = TestRenderer.create( element );
    return testRenderer.root;
}; 
```
Not that much of the work needed :)

Reply:
Actually, it replaces `mount` as far as I can see. Interesting 🤔

Reply:
Would be good to explore removing enzyme in lieu of this approach in a separate PR. The lack of React 16 support is frustrating. My only concern is whether or not it makes our tests run significantly slower.

Reply:
Yes, definitely something we should explore further. Actually, `enzyme` bring lots of performance penalty on its own. See:
https://github.com/WordPress/packages/blob/master/packages/jest-preset-default/scripts/setup-test-framework.js#L3-L6